### PR TITLE
Fix #4412: os_security_group_rule support numbered protocols

### LINKED ISSUE #4412: os_security_group_rule support numbered protocols
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

os_security_group_rule
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 57fca2dde2) last updated 2016/08/10 09:50:16 (GMT -400)
  lib/ansible/modules/core: (detached HEAD a216ef210b) last updated 2016/08/10 09:50:43 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 39153ea154) last updated 2016/08/10 09:50:43 (GMT -400)
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

None
##### OS / ENVIRONMENT

Ubuntu 14.04
##### SUMMARY

I am trying to take advantage of the `protocol: 112` functionality of neutron to allow VRRP traffic between two instances that share a vIP. Since VRRP is a numbered protocol and with a legitimate use case, there should be an option to support numbered protocols in this module.

Openstack heats supports this with a simple:
`- protocol: 112`
##### STEPS TO REPRODUCE

Create a security group rule such as:

```
- os_security_group_rule:
    security_group: loadbalancer_sg
    protocol: 112
    remote_group: loadbalancer-node_sg
```
##### EXPECTED RESULTS

The security group rule is created, as this is a feature of openstack.
##### ACTUAL RESULTS

The module failed to add as it can only handle tcp,udp,icmp, or None

```
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "value of protocol must be one of: None,tcp,udp,icmp, got: 112"}
```

### PR DESCRIPTION
**Pull request by davidjpeacock**
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

cloud/openstack/os_security_group_rule.py
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

Bugfix correctly points out lack of VRRP support by way of numbered protocol.  This commit adds this functionality, including example usage in the documentation.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @davidjpeacock. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_security_group_rule.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
    protocol:
       description:
         - IP protocol
-      choices: ['tcp', 'udp', 'icmp', None]
+      choices: ['tcp', 'udp', 'icmp', 112, None]
```
Comment (ryansb) [inline cloud/openstack/os_security_group_rule.py:42]: In a followup PR can you add to the description here what `112` does?

Comment (davidjpeacock) [reply cloud/openstack/os_security_group_rule.py:42]: Certainly will; thanks Ryan.

**Comment by indispeq** [comment]
Hello, this is in fact not fixed at least not in the most current devel, there is a typo. I addressed it in https://github.com/ansible/ansible-modules-core/pull/4750 , it works with the apostrophes around the 112

**Pull request merged**